django/chat/llm.py

```diff
@@ -0,0 +1,252 @@
+from django.conf import settings
+
+import tiktoken
+from llama_index.core import PromptTemplate, ServiceContext, VectorStoreIndex
+from llama_index.core.callbacks import CallbackManager, TokenCountingHandler
+from llama_index.core.response_synthesizers import CompactAndRefine, TreeSummarize
+from llama_index.core.retrievers import QueryFusionRetriever
+from llama_index.core.vector_stores.types import MetadataFilter, MetadataFilters
+from llama_index.embeddings.azure_openai import AzureOpenAIEmbedding
+from llama_index.llms.azure_openai import AzureOpenAI
+from llama_index.vector_stores.postgres import PGVectorStore
+
+from otto.models import Cost, User
+
+
+class OttoLLM:
```

Comment:
"LLM" is a slight misnomer since this also includes embedding / vector store functions, but I wasn't sure what else to call it. And it is far more convenient to have them bundled together here.